pkg/status/render/render.go

```diff
@@ -229,9 +234,19 @@ func renderStatusTemplate(w io.Writer, templateName string, stats interface{}) e
 	return t.Execute(w, stats)
 }
 
+// ParseHTMLTemplate the HTML template with the data provided
+func ParseHTMLTemplate(w io.Writer, templateName string, stats interface{}) error {
+	tmpl, tmplErr := templatesFS.ReadFile(path.Join("templates", templateName))
+	if tmplErr != nil {
+		return tmplErr
+	}
+	t := htmlTemplate.Must(htmlTemplate.New(templateName).Funcs(htmlfmap).Parse(string(tmpl)))
```

Comment:
`Must` will panic on error, which feels wrong because this function will sometimes return an error. Instead, if `Parse` returns an error just return that as well.

Reply:
That's interesting. Some questions I would have:
1) If there a `recover` somewhere that turns the panic into a rendered error? I haven't investigated at all.
2) What would the behavior be if this returned an error instead of panicing? Would that error still get rendered in an error section? If so I think it would be an upgrade to remove the possibility of a panic.
3) Because there methods are exported, it's more likely that other callers might use this code without realizing it might panic, especially because the signature shows that the function can return an error.

Reply:
> What would the behavior be if this returned an error instead of panicing? Would that error still get rendered in an error section? If so, it would be an upgrade to remove the possibility of a panic.

The render package calls the `renderErrors` functions with all the accumulated errors:

https://github.com/DataDog/datadog-agent/blob/7d485d96d0b45889d22206a1d737c160ab4374a8/pkg/status/render/render.go#L72

https://github.com/DataDog/datadog-agent/blob/7d485d96d0b45889d22206a1d737c160ab4374a8/pkg/status/render/render.go#L118

> If there a recover somewhere that turns the panic into a rendered error? I haven't investigated at all.

I don't think there is a `recover` I think it just doesn't fail to parse the template.

> Because there methods are exported, it's more likely that other callers might use this code without realizing it might panic, especially because the signature shows that the function can return an error.

That is a fair point 😄 

I can change this package in this PR, but I can see that in other places of the agent, we use this pattern:

![image](https://github.com/DataDog/datadog-agent/assets/4672858/d5dd8210-d6c6-460b-9930-54d8888dd81d)

I will fix the other cases in a later PR

Reply:
Done [e0ac5ce](https://github.com/DataDog/datadog-agent/pull/21881/commits/e0ac5ced6acc8487926a37330598628066496daf)

Reply:
Yeah, I don't think this is urgent to fix, I'm okay with merging as is.